### PR TITLE
feat: initialize  Github Pages for hosting Javadocs

### PR DESCRIPTION
**Pull request by alicejli**
https://github.com/googleapis/java-cloud-bom/issues/5950

This should initialize the Github Pages site on this repo with some mock data.

### CODE REVIEW

**Comment by alicejli** [comment]
@suztomo I wasn't sure if there was anything special we needed to do in terms of adding another workflow to this repo (to publish the Github Pages site), so adding you as a reviewer.

File: .github/workflows/hugo.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    shell: bash
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    env:
+      HUGO_VERSION: 0.111.3
+    steps:
+      - name: Install Hugo CLI
```
Comment (meltsufin) [inline .github/workflows/hugo.yaml:38]: What's the purpose of Hugo? Can we just host the pure javadocs on gh-pages?

File: my-new-site/config.toml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+# Used in the "version-banner" partial to display a version number for the
+# current doc set.
+version = "0.0"
+
+# A link to latest version of the docs. Used in the "version-banner" partial to
+# point people to the main doc site.
+url_latest_version = "https://example.com"
+
+# Repository configuration (URLs for in-page links to opening issues and suggesting changes)
+github_repo = "https://github.com/alicejli/javadocs-test-all-modules"
```
Comment (burkedavison) [inline my-new-site/config.toml:108]: Could you create a fork of the java-cloud-bom repo (.../alicejli/java-cloud-bom) and develop this workflow in the fork? I think that'd allow us to get a better understanding of what is being provided, and when it's ready to merge into this one.

Comment (alicejli) [reply my-new-site/config.toml:108]: That's a good point. Let me recreate this PR in that fork and I'll tag you there.

File: .github/workflows/hugo.yaml (hunk 1 above)
Comment (meltsufin) [reply .github/workflows/hugo.yaml:38]: Got it, thanks for the clarification!

**Comment by alicejli** [comment]
marking as `do not merge` - will open another PR from my fork when its ready to go into this.

**Comment by alicejli** [comment]
Closing this as we went a different direction with https://github.com/googleapis/java-cloud-bom/pull/6083

**Pull request closed without merging**